Framework/DataHandling/src/LoadEventNexus.cpp

```diff
@@ -1081,56 +1086,52 @@ void LoadEventNexus::deleteBanks(EventWorkspaceCollection_sptr workspace,
 void LoadEventNexus::createSpectraMapping(
     const std::string &nxsfile, const bool monitorsOnly,
     const std::vector<std::string> &bankNames) {
-  bool spectramap = false;
   m_specMin = getProperty("SpectrumMin");
   m_specMax = getProperty("SpectrumMax");
   m_specList = getProperty("SpectrumList");
 
-  // set up the
   if (!monitorsOnly && !bankNames.empty()) {
-    std::vector<IDetector_const_sptr> allDets;
-
+    const auto &componentInfo = m_ws->getSingleHeldWorkspace()->componentInfo();
+    std::vector<SpectrumDefinition> spectrumDefinitions;
     for (const auto &bankName : bankNames) {
-      // Only build the map for the single bank
-      std::vector<IDetector_const_sptr> dets;
-      m_ws->getInstrument()->getDetectorsInBank(dets, bankName);
+      const auto &bank = m_ws->getInstrument()->getComponentByName(bankName);
+      std::vector<size_t> dets;
+      if (bank) {
+        const auto bankIndex = componentInfo.indexOf(bank->getComponentID());
+        dets = componentInfo.detectorsInSubtree(bankIndex);
```

Comment:
It's possible you could avoid the allocation here and use the `ComponentInfo` `Range`s (see my BoundingBox pr). That would mean moving the `SpectrumDefinition`s generation internally into `ComponentInfo`. Maybe though that's a dependency that we don't want to introduce?

Reply:
I don't think this is an issue. It used to create a vector of shared pointers to `Detector`, which was much much more work. So the new code should be at least as efficient.